### ISSUE TITLE
Weird pixelated edges on amdgpu on linux

### ISSUE THREAD

**Issue by WebFreak001** [ppy/osu#3118]
![https://wfr.moe/fDWLft.png](https://wfr.moe/fDWLft.png)
![https://wfr.moe/fDWDmH.png](https://wfr.moe/fDWDmH.png)
![https://wfr.moe/fDs6A8.png](https://wfr.moe/fDs6A8.png)
![https://wfr.moe/fDsQy0.png](https://wfr.moe/fDsQy0.png)
![https://wfr.moe/fDszws.png](https://wfr.moe/fDszws.png)
![https://wfr.moe/fDBLiG.png](https://wfr.moe/fDBLiG.png)
![https://wfr.moe/fDBDJG.png](https://wfr.moe/fDBDJG.png)

```
Linux archlinux 4.17.5-1-ARCH #1 SMP PREEMPT Sun Jul 8 17:27:31 UTC 2018 x86_64 GNU/Linux

OpenGL vendor string: X.Org
OpenGL renderer string: AMD Radeon (TM) RX 480 Graphics (POLARIS10, DRM 3.25.0, 4.17.5-1-ARCH, LLVM 6.0.1)
OpenGL core profile version string: 4.5 (Core Profile) Mesa 18.1.4
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 3.1 Mesa 18.1.4
OpenGL shading language version string: 1.40
OpenGL context flags: (none)
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.1 Mesa 18.1.4
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
OpenGL ES profile extensions:
```

**Comment by WebFreak001** [comment]
If you remove the mipmap bias in the texture2D call in sh_TextureRounded this seems to be resolved:
![https://wfr.moe/fD3D3t.png](https://wfr.moe/fD3D3t.png)
![https://wfr.moe/fDJ6ca.png](https://wfr.moe/fDJ6ca.png)
![https://wfr.moe/fDJQ8w.png](https://wfr.moe/fDJQ8w.png)
![https://wfr.moe/fDJz7M.png](https://wfr.moe/fDJz7M.png)
![https://wfr.moe/fDpLiv.png](https://wfr.moe/fDpLiv.png)
![https://wfr.moe/fDpDpS.png](https://wfr.moe/fDpDpS.png)
![https://wfr.moe/fDC6lE.png](https://wfr.moe/fDC6lE.png)

However some stuff looks a bit blurry now
Before: ![https://wfr.moe/fDCznO.png](https://wfr.moe/fDCznO.png)
After: ![https://wfr.moe/fDCQjx.png](https://wfr.moe/fDCQjx.png)

**Comment by peppy** [comment]
Did you end up trying to expand the white texture area to see if that helps?

**Comment by WebFreak001** [comment]
yes it didn't change anything. I don't quite see how it would have fixed it?

**Comment by peppy** [comment]
We isolated it to the mipmap creation, no?

Which means that increasing the padding of white around the actually used pixel (in the centre of the whiteTex) would have a lower chance of being incorrectly mipmapped when larger.

I started to make a [branch](https://github.com/peppy/osu-framework/tree/white-pixel-testing) for this but have run into an issue when moving the white pixel away from the top-left-most pixel. Will need to figure out why this is happening to move forward here.
